### PR TITLE
stdlib: Dividing range for linear multicore.

### PR DESCRIPTION
**Pull request by mahyarsamani**
This patch, changes the way memory ranges are devided when using multiple cores for linear traffic. The current state assigns the same range to multiple linear generators so all the cores start generating the same trace. This patch devides the overall range assigned to the generator ([min_addr:max_addr]) between the cores.

Change-Id: I49f69b3d61b590899f8d54ee3be997ad22d7fa9b

### CODE REVIEW

File: src/python/gem5/components/processors/abstract_generator.py [hunk 1]
```diff
@@ -35,6 +35,20 @@
 from typing import List
 
 
+def _devide_memory_ranges(min_addr, max_addr, num_partitions):
```
Comment (BobbyRBruce) [inline src/python/gem5/components/processors/abstract_generator.py]: Two TODOs here:

- Rename `_divide_memory_ranges`
- Add typing (ints for the parameter and the return type should be `List[int]` if I'm reading this correctly.

**Review by BobbyRBruce — CHANGES_REQUESTED** [review]

File: src/python/gem5/components/processors/abstract_generator.py (hunk 1 above)
Comment (mahyarsamani) [reply src/python/gem5/components/processors/abstract_generator.py]: Done.

**Comment by mahyarsamani** [comment]
Is there anything I should do on my end? I see "changes requested" but I can't find anything apart from Bobby's comments.

**Review by BobbyRBruce — APPROVED** [review]

**Comment by BobbyRBruce** [comment]
> Is there anything I should do on my end? I see "changes requested" but I can't find anything apart from Bobby's comments.

No, you're good. I just needed to re-review the changes.

**Comment by BobbyRBruce** [comment]
@mahyarsamani : Can you see if you can reproduce these failing tests on your end?

**Pull request merged**